garden_ai/client.py

```diff
@@ -295,9 +299,29 @@ def create_pipeline(
         return Pipeline(**data)
 
     def register_model(self, local_model: LocalModel) -> RegisteredModel:
-        registered_model = upload_to_model_registry(local_model)
-        local_data.put_local_model(registered_model)
-        return registered_model
+        if "S3_UPLOAD_FEATURE_FLAG" in os.environ:
+            try:
+                # Create directory in MLModel format
+                model_directory = stage_model_for_upload(local_model)
+                # Push contents of directory to S3
+                upload_mlmodel_to_s3(model_directory, local_model, self.backend_client)
+            finally:
+                try:
+                    clear_mlflow_staging_directory()
+                except Exception as e:
+                    logger.error(
+                        "Could not clean up model staging directory. Check permissions on ~/.garden/mlflow"
+                    )
+                    logger.error("Original exception: " + str(e))
+                    # We can still proceed, there is just some cruft in the user's home directory.
+                    pass
+            registered_model = RegisteredModel(**local_model.dict(), version="1")
```

Comment:
Another note for reviewers: I'm keeping the RegisteredModel class as is in this PR but I plan on changing it when I can delete the old code in the follow-up.